app/src/main/java/com/example/jellyfishandroidexample/RoomViewModel.kt

```diff
@@ -72,34 +71,33 @@ class RoomViewModel(application: Application) :
         emitParticipants()
     }
 
-    override fun onJoinError(metadata: Any) {
-    }
-
-    override fun onRemoved(reason: String) {
+    override fun onConnectError(metadata: Any) {
     }
 
-    override fun onPeerJoined(peer: Peer) {
+    override fun onPeerJoined(peer: Endpoint) {
         mutableParticipants[peer.id] = Participant(
             id = peer.id,
         )
         emitParticipants()
     }
 
-    override fun onPeerLeft(peer: Peer) {
+    override fun onPeerLeft(peer: Endpoint) {
         mutableParticipants.remove(peer.id)
         emitParticipants()
     }
 
+    override fun onPeerUpdated(peer: Endpoint) {}
+
     override fun onTrackReady(ctx: TrackContext) {
         viewModelScope.launch {
-            val participant = mutableParticipants[ctx.peer.id] ?: return@launch
+            val participant = mutableParticipants[ctx.endpoint.id] ?: return@launch
```

Comment:
shouldn't user see `peer` here instead of an `endpoint`? in ts client it's also endpoint so please discuss it with Kamil

Reply:
Afaik there is an idea to support multiple protocols => different names for each and everyone of them. So I think it could be left as it is for now.